chapter/background.md

```diff
@@ -0,0 +1,607 @@
+# Background
+
+This thesis illustrates an approach of implementing a language server for the Nickel language which communicates with its clients, i.e. editors, over the open Language Server Protocol (in the following abbreviated as *LSP*).
+The current chapter provides the background on the technological details of the project.
+As the work presented aims to be transferable to other languages using the same methods, this chapter will provide the means to distinguish the nickel specific implementation details.
+
+The primary technology built upon in this thesis is the language server protocol.
+The first part of this chapter introduces the LSP, its rationale and improvements over classical approaches, technical capabilities and protocol details. 
+The second part is dedicated to Nickel, elaborating on the context and use-cases of the language followed by an inspection of the technical features of Nickel.
+
+## Language Server Protocol
+
```

Comment:
```suggestion

The Language Server Protocol is a JSON-RPC based communication specification comprising an LSP client (i.e. editors) and server (also called language server for simplicity).
The LSP decouples the development of clients and servers, allowing developers to focus on either side.
The LSP defines several capabilities -- standardised functions remotely executed by the language server.
LSP Clients are often implemented as editor extensions facilitating abstraction libraries helping with the interaction with the protocol and editor interface.
Language Servers analyse source code sent by the client and may implement any number of 
capabilities relevant to the language.
Since the LSP is both language and editor independent, the same server implementation can serve all LSP compliant clients eliminating the need to redundantly recreate the same code intelligence for every editor.
  
```